### PR TITLE
Logger.getChild subclass compatible typehint

### PR DESCRIPTION
**Pull request by randolf-scholz**
Proposed fix for https://github.com/python/typeshed/issues/6606

### CODE REVIEW

File: stdlib/logging/__init__.pyi
Comment (Akuli) [inline stdlib/logging/__init__.pyi:63]: It would be better to use `_typeshed.Self` for this:

```suggestion
    def getChild(self: Self, suffix: str) -> Self: ...
```

You will need to add `from _typeshed import Self`.

Comment (randolf-scholz) [reply stdlib/logging/__init__.pyi:63]: So the `bound=` is generally not necessary for this type of operation?

Comment (Akuli) [reply stdlib/logging/__init__.pyi:63]: That's right. `self: T` creates an implicit bound on `T`.

Comment (Akuli) [reply stdlib/logging/__init__.pyi:63]: Maybe I should clarify this a bit more. This is fine:

```
T = TypeVar("T")

class Foo:
    def do_something(self: T) -> T: ...  # T has implicitly bound=Foo

def bar(thing: T) -> T: ...   # no bound on T
```

**Review by Akuli — APPROVED** [review]
Thanks! I'll wait for another maintainer to merge this and confirm whether this is a good idea.

**Comment by JelleZijlstra** [comment]
Ah, I just found python/typing#980 which makes a good argument that this is useful in practice.

**Comment by randolf-scholz** [comment]
> Ah, I just found [python/typing#980](https://github.com/python/typing/discussions/980) which makes a good argument that this is useful in practice.

Well, this is a thread I opened today before opening the issue here. I linked this thread also in the issue. https://github.com/python/typeshed/issues/6606

**Comment by AlexWaygood** [comment]
> > Ah, I just found [python/typing#980](https://github.com/python/typing/discussions/980) which makes a good argument that this is useful in practice.
> 
> Well, this is a thread I opened today before opening the issue here. I linked this thread also in the issue. #6606

It might be good to link to the thread in a comment

File: stdlib/logging/__init__.pyi
Comment (AlexWaygood) [inline stdlib/logging/__init__.pyi:62]:
```suggestion
    def getChild(self: Self, suffix: str) -> Self: ...     # see python/typing#980
```

**Pull request merged**
